### PR TITLE
Recommend "Max Protection" DNS over HTTPS in Firefox

### PR DESCRIPTION
**Pull request by mmrabbani**
Changes proposed in this PR:
Max Protection DOH in Firefox is added to Firefox recommended configuration.
This feature is added to Firefox 118 recently.

More Information:
https://support.mozilla.org/en-US/kb/dns-over-https
- 

- [x] I have disclosed any relevant conflicts of interest in my post.
- [x] I agree to grant Privacy Guides a perpetual, worldwide, non-exclusive, transferable, royalty-free, irrevocable license with the right to sublicense such rights through multiple tiers of sublicensees, to reproduce, modify, display, perform, relicense, and distribute my contribution as part of this project.
- [x] I am the sole author of this work. 
- [x] I agree to the [Community Code of Conduct](https://www.privacyguides.org/en/code_of_conduct/).

### CODE REVIEW

File: docs/desktop-browsers.md [hunk 1]
```diff
@@ -179,6 +179,12 @@ Additionally, the Firefox Accounts service collects [some technical data](https:
 
 This prevents you from unintentionally connecting to a website in plain-text HTTP. Sites without HTTPS are uncommon nowadays, so this should have little to no impact on your day to day browsing.
 
+##### DNS over HTTPS
+
+- [x] Select **Max Protection** DNS over HTTPS
+
+Max protection will always use DNS over HTTPS and a security warning will show if we can’t connect to the secure DNS resolver, or if the secure DNS resolver indicates there are no addresses for the domain you are trying to access. This will prevent ISP and government to monitor your DNS queries. Note that your DNS queries can still be monitored by the DNS server.
```
Comment (jonaharagon) [inline docs/desktop-browsers.md:186]:
```suggestion
Max Protection enforces the use of DNS over HTTPS, and a security warning will show if Firefox can’t connect to your secure DNS resolver, or if your secure DNS resolver says that records for the domain you are trying to access do not exist. This stops the network you're connected to from secretly downgrading your DNS security.
```

File: docs/desktop-browsers.md [hunk 2]
```diff
@@ -179,6 +179,12 @@ Additionally, the Firefox Accounts service collects [some technical data](https:
 
 This prevents you from unintentionally connecting to a website in plain-text HTTP. Sites without HTTPS are uncommon nowadays, so this should have little to no impact on your day to day browsing.
 
+##### DNS over HTTPS
+
+- [x] Select **Max Protection** DNS over HTTPS
```
Comment (jonaharagon) [inline docs/desktop-browsers.md:184]:
```suggestion
If you use a [DNS over HTTPS provider](dns.md):

- [x] Select **Max Protection** and choose a suitable provider
```

**Review by jonaharagon — APPROVED** [review]
Thanks!

**Comment by dngray** [comment]
We probably should mention though, not to touch this if you're using a VPN. It's still better to use your VPN provider's DNS servers than a third party.

**Comment by mmrabbani** [comment]
> We probably should mention though, not to touch this if you're using a VPN. It's still better to use your VPN provider's DNS servers than a third party.

Why it is not recommended to use this feature while using VPN? Firefox will enable ECH by default this month. By using both DOH and ECH, VPN provider cannot monitor which service the client connects. Also The DOH server only sees that the request is from the VPN server and cannot identify the requester.

**Comment by dngray** [comment]
> Firefox will enable ECH by default in the next version this month

I have to wonder though how many servers support ECH. If I remember correctly, that only worked with TLS 1.3. Also ECH isn't yet RFC still draft https://datatracker.ietf.org/doc/draft-ietf-tls-esni/

**Comment by jonaharagon** [comment]
This is why I'm annoyed Firefox *requires* in-browser DOH for ECH to be enabled (for no technical reason) 😠 

> I have to wonder though how many servers support ECH.

The only server that supports ECH and is significant is Cloudflare, but that does represent a notable privacy improvement for 20%+ of internet traffic. From *that* perspective, it may even make sense to recommend using Cloudflare DNS specifically, if you are interested in the privacy benefits of ECH, regardless of VPN use 🤔

**Comment by mmrabbani** [comment]
> This is why I'm annoyed Firefox _requires_ in-browser DOH for ECH to be enabled (for no technical reason) 😠

From Firefox 118, Even the default DNS setting uses secure DNS in regions where it’s available. It is said that ECH will be enabled by default from Firefox 119. So users won't be annoyed anymore

>  but that does represent a notable privacy improvement for 20%+ of internet traffic. 

I agree. I hope other CDN and cloud providers enable ECH by default soon.

**Review by dngray — APPROVED** [review]

**Pull request merged**